src/go/rdctl/cmd/snapshotCreate.go

```diff
@@ -2,44 +2,75 @@ package cmd
 
 import (
 	"fmt"
-	"os/exec"
-	"runtime"
-
 	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/paths"
 	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/snapshot"
+	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
+	"os/exec"
+	"runtime"
+	"strings"
 )
 
+var snapshotDescription string
+
 var snapshotCreateCmd = &cobra.Command{
 	Use:   "create <name>",
 	Short: "Create a snapshot",
 	Args:  cobra.ExactArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
 		cmd.SilenceUsage = true
-		err := wrapSnapshotOperation(createSnapshot)(cmd, args)
-		return exitWithJsonOrErrorCondition(err)
+		return exitWithJsonOrErrorCondition(createSnapshot(cmd, args))
 	},
 }
 
 func init() {
 	snapshotCmd.AddCommand(snapshotCreateCmd)
-	snapshotCreateCmd.Flags().BoolVarP(&outputJsonFormat, "json", "", false, "output json format")
+	snapshotCreateCmd.Flags().BoolVar(&outputJsonFormat, "json", false, "output json format")
+	snapshotCreateCmd.Flags().StringVar(&snapshotDescription, "description", "", "snapshot description")
 }
 
-func createSnapshot(_ *cobra.Command, args []string) error {
+func createSnapshot(cmd *cobra.Command, args []string) error {
 	appPaths, err := paths.GetPaths()
 	if err != nil {
 		return fmt.Errorf("failed to get paths: %w", err)
 	}
 	manager := snapshot.NewManager(appPaths)
-	if _, err := manager.Create(args[0]); err != nil {
-		return fmt.Errorf("failed to create snapshot: %w", err)
+	if err := manager.ValidateName(args[0]); err != nil {
+		return err
+	}
+	err = wrapSnapshotOperation(cmd, appPaths, false, func() error {
+		if _, err := manager.Create(args[0], snapshotDescription); err != nil {
+			return fmt.Errorf("failed to create snapshot: %w", err)
+		}
+		return nil
+	})
+	if err != nil {
+		return err
 	}
+	if runtime.GOOS != "darwin" {
+		return nil
+	}
+
 	// exclude snapshots directory from time machine backups if on macOS
-	if runtime.GOOS == "darwin" {
-		cmd := exec.Command("tmutil", "addexclusion", appPaths.Snapshots)
-		if err := cmd.Run(); err != nil {
-			return fmt.Errorf("failed to add exclusion to TimeMachine: %w", err)
+	execCmd := exec.Command("tmutil", "isexcluded", appPaths.Snapshots)
```

Comment:
Why add all of this complexity when `tmutil addexclusion` is idempotent? This is a premature optimization, no?